src/HttpCache/TagsInvalidator/Banner.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\HttpCache\TagsInvalidator;
+
+use ApiPlatform\HttpCache\TagsInvalidatorInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class Banner implements TagsInvalidatorInterface
```

Comment:
The `BAN` invalidation method is quite specific to Varnish and we probably don't want this to be too generic. Indeed, we recommend to use surrogate keys instead as it's less error-prone in its implementation then using regular expressions.

Reply:
Specific to Varnish does not mean it cannot be configurable.

Even if you recommend surrogate keys this invalidator exists so it has to continue existing.

Reply:
Only Varnish is using `Ban`-wording. I don't see the benefits of keeping the Banner AND the Purger.